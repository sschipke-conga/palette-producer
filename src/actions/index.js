

export const setUser = user => ({
  type: 'SET_USER',
  user
});

export const setAllProjects = allProjects => ({
  type: 'SET_PROJECTS',
  allProjects
});

export const setCurrentPalette = palette => ({
  type: 'SET_CURRENT_PALETTE',
  palette
});

export const setCurrentPaletteInfo = info => ({
  type: 'SET_CURRENT_PALETTE_INFO',
  info
});

export const setCurrentProjectInfo = info => ({
  type: 'SET_CURRENT_PROJECT_INFO',
  info
});

export const setAllPalettes = allPalettes => ({
  type: 'SET_PALETTES',
  allPalettes
});

export const toggleMenu = () => ({
  type: 'TOGGLE_MENU'
})

export const removePalette = (id) => ({
  type: 'REMOVE_PALETTE',
  id: parseInt(id)
});