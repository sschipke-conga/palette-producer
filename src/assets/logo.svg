<svg id="Layer_1" data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" width="316.9996" height="316.9996" viewBox="0 0 316.9996 316.9996"><defs><style>.cls-1{fill:#fff;}.cls-1,.cls-2,.cls-3,.cls-4,.cls-5{stroke:#000;stroke-miterlimit:10;stroke-width:10px;}.cls-2{fill:#fbb03b;}.cls-3{fill:#f15a24;}.cls-4{fill:#d4145a;}.cls-5{fill:#3c4892;}</style></defs><title>Artboard 5</title><path class="cls-1" d="M31.62037,143.03632c.33314-55.44444,42.19532-108.237,95.8371-120.86067C185.4136,8.53673,248.78037,35.47176,273.57154,85.89533c31.15582,63.36876,1.52388,129.99623-69.57942,114.45468-13.68126-2.99042-12.8967-5.45845-31.15819-10.9585-23.80811-7.1706-24.20607,3.48171-14.73851,56.5,5,28,7,39-8,49C111.92072,320.34136,31.22091,209.51685,31.62037,143.03632Z"/><circle class="cls-2" cx="75.01795" cy="138.12396" r="22.92254"/><circle class="cls-3" cx="116.7926" cy="76.84227" r="22.92254"/><circle class="cls-4" cx="195.10246" cy="76.8141" r="22.92254"/><circle class="cls-5" cx="230.17288" cy="147.88452" r="22.92254"/></svg>