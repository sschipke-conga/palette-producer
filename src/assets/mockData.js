export const mockPalettes = [
  {
    id: 30,
    project_id: 35,
    name: "My Second Palette",
    color1: "#e5349d",
    color2: "#bdaac5",
    color3: "#5a9def",
    color4: "#2705b2",
    color5: "#309a6e",
    created_at: "2020-01-21T19:09:20.992Z",
    updated_at: "2020-01-21T19:09:20.992Z",
  },
  {
    id: 33,
    project_id: 35,
    name: "New one now",
    color1: "#f36d6b",
    color2: "#72cc20",
    color3: "#f11a21",
    color4: "#8eb4e8",
    color5: "#b08980",
    created_at: "2020-02-15T00:06:46.617Z",
    updated_at: "2020-02-15T00:06:46.617Z",
  },
  {
    id: 40,
    project_id: 35,
    name: "This time it should work",
    color1: "#57101d",
    color2: "#324086",
    color3: "#5a789c",
    color4: "#959191",
    color5: "#18a647",
    created_at: "2020-02-19T03:52:55.467Z",
    updated_at: "2020-02-19T03:52:55.467Z",
  },
  {
    id: 45,
    project_id: 37,
    name: "My Second Palette",
    color1: "#e5349d",
    color2: "#bdaac5",
    color3: "#5a9def",
    color4: "#2705b2",
    color5: "#309a6e",
    created_at: "2020-01-21T19:09:20.992Z",
    updated_at: "2020-01-21T19:09:20.992Z"
  },
  {
    id: 43,
    project_id: 37,
    name: "This time it should work",
    color1: "#57101d",
    color2: "#324086",
    color3: "#5a789c",
    color4: "#959191",
    color5: "#18a647",
    created_at: "2020-02-19T03:52:55.467Z",
    updated_at: "2020-02-19T03:52:55.467Z"
  }
]

export const mockProjects = [
  {
    id: 30,
    name: "My First Project",
  },
  {
    id: 33,
    name: "Second Projects",
  },
  {
    id: 40,
    name: "Third Project"
  }
]

export const mockUser = {
  user_id: 34,
  username: 'Gary'
}

export const mockCurrentPalette = [
  {
    name: 'color1',
    hexCode: '#FFFFFF',
    isLocked: false
  },
  {
    name: 'color2',
    hexCode: '#FFFFFF',
    isLocked: false
  },
  {
    name: 'color3',
    hexCode: '#FFFFFF',
    isLocked: false
  },
  {
    name: 'color4',
    hexCode: '#FFFFFF',
    isLocked: false
  },
  {
    name: 'color5',
    hexCode: '#FFFFFF',
    isLocked: false
  }
]