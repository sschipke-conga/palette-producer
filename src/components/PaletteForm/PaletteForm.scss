
.palette-form {
  .update-palette-div {
    margin: 15px auto;
    width: 40%;
    justify-content: space-between;
    h4 {
      font-size: 2em;
    }
    .update-palette-buttons {
      display: flex;
      justify-content: space-between;
    }
    span {
      font-weight: normal;
    }
  }
  button {
    background: #FDF7DB;
    border: 1px solid purple;
    border-radius: 5px;
    font-size: 1.5em;
    margin: 10 10px;
    height: 40px;
    width: 40%;
    &:hover {
      color: white;
      background: purple;
    }
  }
  .create-palette-button {
    border: 1px solid red;
    color: maroon;
    &:hover {
      background: red;
      color: white;
      border: white
    }
  }
    .message {
      display: flex;
      align-items: center;
      flex-direction: column;
      justify-content: center;
      height: 50%;
      h5 {
        color: purple;
        font-size: 1.5em;
        margin-bottom: 20px;
      }
      button {
        background: purple;
        color: white;
        &:hover {
          background: white;
          color: purple;
        }
      }
  }
}

.editor-form {
  width: 40%;
  margin: 0 auto;
  .project-editor-div {
  display: flex;
  }
  label {
    font-size: 1.3em;
  }
  input {
    font-size: 1.2em;
    background: #FDF7DB;
    border-bottom: 1px solid purple;
    &::placeholder {
      color: rgba(128,0,128, 0.6)
    }
  }
  padding: 0;
  select {
    background: #FDF7DB;
    border: 1px solid purple;
    border-radius: 5px;
    font-size: 1.3em;
    margin-right: 10px;
  }

}