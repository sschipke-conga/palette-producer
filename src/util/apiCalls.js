export const createNewUser = async user => {
  let url = `${process.env.REACT_APP_PALETTE_PRODUCER_BACKEND_BASE_URL}/api/v1/signup`;
  let options = {
    method: "POST",
    body: JSON.stringify(user),
    headers: {
      "Content-Type": "application/json"
    }
  };
  let res = await fetch(url, options);

  if (res.status === 500) {
    throw Error("This username already taken");
  }

  if (!res.ok) {
    throw Error("Woops! Something went wrong");
  }

  return res.json();
};

export const loginUser = async user => {
  let url = `${process.env.REACT_APP_PALETTE_PRODUCER_BACKEND_BASE_URL}/api/v1/login`;
  let options = {
    method: "POST",
    body: JSON.stringify(user),
    headers: {
      "Content-Type": "application/json"
    }
  };
  let res = await fetch(url, options);

  if (res.status === 401) {
    throw Error("Username or password is incorrect");
  }

  if (!res.ok) {
    throw Error("Woops! Something went wrong");
  }

  return res.json();
};
