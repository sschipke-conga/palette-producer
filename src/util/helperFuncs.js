export const generateRandomHex = () => {
  let hexCode = "#" + Math.floor(Math.random() * 16777215).toString(16);
  if (hexCode.length < 7) {
    return generateRandomHex();
  }
  return hexCode;
};
