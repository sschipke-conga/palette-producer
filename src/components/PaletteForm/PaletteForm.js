
import React, { Component } from "react";
import { Redirect } from "react-router-dom";
import { loginUser } from "../../util/apiCalls";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import {savePalette, saveProject} from '../../util/apiCalls'
import { setCurrentPalette, setAllPalettes } from "../../actions/index";
import "./PaletteForm.test";



class PaletteForm extends Component {
  constructor() {
    super();
    this.state = {
      paletteName: '',
      projectName: '',
      projectId: '',
      error: ''
    };
  }

  handleChange = e => {
    this.setState({ [e.target.id]: e.target.value });
  };

  handleSubmit = async e => {
    e.preventDefault();
    this.savePalette()
    e.target.reset()
  };

  displayProjectOptions = () => {
    return this.props.allProjects.map((project, index) => {
    return <option key={'Option' + project.id}value={project.id}>{project.name}</option>
    })
  }

  savePalette = async () => {
    const {paletteName, projectId, projectName} = this.state
    const {currentPalette, user} = this.props
    const paletteToPost = {
      color1: currentPalette[0].hexCode,
      color2: currentPalette[1].hexCode,
      color3: currentPalette[0].hexCode,
      color4: currentPalette[0].hexCode,
      color5: currentPalette[0].hexCode,
      name: paletteName,
      project_id: projectId
    };
    console.log(paletteToPost)
    if(projectId === 'CreateNew' ) {
      try {
        const res = await saveProject({name: projectName, user_id: user.user_id})
        paletteToPost.project_id = res.id
        savePalette(paletteToPost)
      } catch ({error}) {
        console.error(error)
      }
    } else {
      try {
        savePalette(paletteToPost)
      } catch ({error}) {
        console.error(error)
      }
    }

  }

  render() {
    const { paletteName, projectName, error, isLoggedIn, projectId,} = this.state;

    let loginErrrorClass = error ? "input-error" : "";
    return (
      <form onSubmit={this.handleSubmit}>
        {error ? <p className="error-message">{error}</p> : null}
        <div className="form-label-input-div">
          <label htmlFor="password">Project</label>
          <select
            value={projectId}
            onChange={this.handleChange}
            id="projectId"
            required
          >
            <option value="" defaultValue>
              Please select a project
            </option>
            <option value="CreateNew" defaultValue>
              Create a new project
            </option>
            {this.displayProjectOptions()}
          </select>
        </div>
        {projectId === "CreateNew" && (
          <div className="form-label-input-div">
            <label htmlFor="paletteName">Project Name</label>
            <input
              type="text"
              placeholder="New Project's Name"
              id="projectName"
              required
              value={projectName}
              onChange={this.handleChange}
            />
          </div>
        )}
        <div className="form-label-input-div">
          <label htmlFor="paletteName">Palette Name</label>
          <input
            type="text"
            className={loginErrrorClass}
            placeholder="Enter your username"
            id="paletteName"
            required
            value={paletteName}
            onChange={this.handleChange}
          />
        </div>
        <button type="submit">Save Palette</button>
      </form>
    );
  }
}

export const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      setCurrentPalette,
      setAllPalettes
    },
    dispatch
  );

export const mapStateToProps = state => ({
  allPalettes: state.allPalettes,
  allProjects: state.allProjects,
  currentPalette: state.currentPalette,
  user: state.user,
  isMenuActive: state.isMenuActive
});

export default connect(mapStateToProps, mapDispatchToProps)(PaletteForm);

