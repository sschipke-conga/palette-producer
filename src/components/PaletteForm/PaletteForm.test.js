import React from 'react';
import { PaletteForm, mapDispatchToProps, mapStateToProps } from './PaletteForm';
import { shallow } from 'enzyme';
import { savePalette, saveProject, updatePalette } from '../../util/apiCalls';
import {mockPalettes, mockUser, mockProjects, mockCurrentPalette} from '../../assets/mockData'
import { setCurrentPalette, addPalette, addProject, resetSelectedPalette, resetSelectedProject, resetCurrentPalette, updateStoredPalette } from '../../actions/index'

jest.mock('../../util/apiCalls')

const mockSetCurrentPalette = jest.fn()
const mockAddProject = jest.fn()
const mockAddPalette = jest.fn()
const mockResetSelectedPalette = jest.fn()
const mockResetSelectedProject = jest.fn()
const mockResetCurrentPalette = jest.fn()
const mockUpdateStoredPalette = jest.fn()


describe('PaletteForm', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<PaletteForm
      setCurrentPalette={mockSetCurrentPalette}
      addPalette={mockAddPalette}
      addProject={mockAddProject}
      resetCurrentPalette={mockResetCurrentPalette}
      resetSelectedPalette={mockResetSelectedPalette}
      resetSelectedProject={mockResetSelectedProject}
      allProjects={mockProjects}
      user={mockUser}
      currentPalette={mockCurrentPalette}
      selectedPaletteInfo={{name: null, id: null, project_id:null}}
      selectedProjectInfo={mockProjects[2]}
      updateStoredPalette={mockUpdateStoredPalette}
      />)
    })
    afterEach(() => {
      jest.clearAllMocks()
    })
  it('should match the initial snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });
  describe('handleChange', () => {
    it('should update state if a user selects a project', () => {
      const mockEvent = { target: { id: 'projectId', value: '5' } };
      expect(wrapper.state('projectId')).toEqual('');
      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state('projectId')).toEqual(mockEvent.target.value)
    })
    it('should update state when a user enters a palette name', () => {
      const mockEvent = { target: { id: 'paletteName', value: 'Palette Name' } };
      expect(wrapper.state('paletteName')).toEqual('');
      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state('paletteName')).toEqual(mockEvent.target.value)
    })
    it('the projectId is CreateNew it should allow a user to enter a name for their project', () => {
        const mockEvent = { target: { id: 'projectName', value: 'Project Name' } };
        expect(wrapper.state('projectName')).toEqual('');
        wrapper.instance().setState({projectId: 'CreateNew'})
        expect(wrapper.state('projectId')).toEqual('CreateNew')
        wrapper.instance().handleChange(mockEvent);
        expect(wrapper.state('projectName')).toEqual(mockEvent.target.value)
      })
    })
    describe('handleSubmit', () => {
      it('should prevent the default action when the form is submitted', () => {
        const mockEvent = { preventDefault: jest.fn(),};
        wrapper.find('form').simulate('submit', mockEvent);
        expect(mockEvent.preventDefault).toHaveBeenCalled()
      })
      it('should call savePalette when called', () => {
        wrapper.instance().savePalette = jest.fn()
        wrapper.instance().forceUpdate()
        let mockEvent = { preventDefault: jest.fn() }
        wrapper.instance().setState({ projectId: '5', paletteName: 'Testing', projectName:'Test project' });
        wrapper.instance().handleSubmit(mockEvent);
        expect(wrapper.instance().savePalette).toHaveBeenCalledWith();
      })
    })
    describe('displayProjectOptions', () => {
      it('should return an array of options for all the projects', () => {
        const result = wrapper.instance().displayProjectOptions()
        expect(result.length).toEqual(mockProjects.length)
      })
    })
    describe('savePalette', () => {
      saveProject.mockImplementation(() => Promise.resolve({id: 67, name:'Fake project'}))
      savePalette.mockImplementation(() => Promise.resolve({id: 88, project_id:67, name: 'Bigsby'}))
      it('if projectId is CreateNew it should call saveProject, savePalette, addPalette and reset', async () => {
        wrapper.instance().reset = jest.fn()
        wrapper.instance().forceUpdate()
        wrapper.setState({projectId: 'CreateNew', projectName: 'Fake project', paletteName: 'Bigsby'})
        await wrapper.instance().savePalette()
        expect(saveProject).toHaveBeenCalledWith({name: 'Fake project', user_id: mockUser.user_id})
        expect(savePalette).toHaveBeenCalled()
        expect(mockAddProject).toHaveBeenCalled()
        expect(mockAddPalette).toHaveBeenCalled()
        expect(wrapper.instance().reset).toHaveBeenCalled()
      })
      it('if projectId is and existing project it should only call savePalette, addPalette and reset', async () => {
        wrapper.instance().reset = jest.fn()
        wrapper.instance().forceUpdate()
        wrapper.setState({ projectId: '8', paletteName: 'Debra' })
        await wrapper.instance().savePalette()
        expect(saveProject).not.toHaveBeenCalled()
        expect(savePalette).toHaveBeenCalled()
        expect(mockAddProject).not.toHaveBeenCalled()
        expect(mockAddPalette).toHaveBeenCalled()
        expect(wrapper.instance().reset).toHaveBeenCalled()
      })
    })
    describe('updateSelectedPalette', () => {
      updatePalette.mockImplementation(() => Promise.resolve(mockPalettes[2]))
      it('should call updatePalette and updateStorePalette', async () => {
        await wrapper.instance().updateSelectedPalette()
        expect(updatePalette).toHaveBeenCalled()
        expect(mockUpdateStoredPalette).toHaveBeenCalled()
      })
      it('should set the state of isUpdateSuccess if the palette is updated successfully', async () => {
        expect(wrapper.state('isUpdateSuccess')).toBe(false)
        await wrapper.instance().updateSelectedPalette()
        expect(wrapper.state('isUpdateSuccess')).toBe(true)
      })
      it('should NOT set the state of isUpdateSuccess if the palette is not updated successfully or call updateStoredPalette', async () => {
        updatePalette.mockImplementation(() => Promise.reject({error:'Test Woops'}))
        expect(wrapper.state('isUpdateSuccess')).toBe(false)
        await wrapper.instance().updateSelectedPalette()
        expect(wrapper.state('isUpdateSuccess')).toBe(false)
        expect(mockUpdateStoredPalette).not.toHaveBeenCalled()
      })
    })
    describe('reset', () => {
      it('should reset the state of the PaletteForm', () => {
        const initialState = {
          paletteName: '',
          projectName: '',
          projectId: '',
          error: '',
          isUpdateSuccess: false
        }
        expect(wrapper.state()).toEqual(initialState)
        wrapper.instance().setState({
          projectId: '5',
          projectName: 'Test Name',
          paletteName: 'Palette name test',
          error: 'Test error',
          isUpdateSuccess: true
        })
        expect(wrapper.state()).toEqual({
          projectId: '5',
          projectName: 'Test Name',
          paletteName: 'Palette name test',
          error: 'Test error',
          isUpdateSuccess: true
        })
        wrapper.instance().reset()
        expect(wrapper.state()).toEqual(initialState)
      })
    })
    describe('altSnapShot1', () => {
      let wrapper;
      beforeEach(() => {
        wrapper = shallow(<PaletteForm
          setCurrentPalette={mockSetCurrentPalette}
          addPalette={mockAddPalette}
          addProject={mockAddProject}
          resetCurrentPalette={mockResetCurrentPalette}
          resetSelectedPalette={mockResetSelectedPalette}
          resetSelectedProject={mockResetSelectedProject}
          allProjects={mockProjects}
          user={mockUser}
          currentPalette={mockCurrentPalette}
          selectedPaletteInfo={{ name: 'Fake Name', id: 5, project_id: 22 }}
          selectedProjectInfo={mockProjects[2]}
          updateStoredPalette={mockUpdateStoredPalette}
        />)
      })
      it('should match the alternate snapshot if there is a selectedPalette', () => {
        expect(wrapper).toMatchSnapshot()
      });
    })
  describe('altSnapShot2', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<PaletteForm
        setCurrentPalette={mockSetCurrentPalette}
        addPalette={mockAddPalette}
        addProject={mockAddProject}
        resetCurrentPalette={mockResetCurrentPalette}
        resetSelectedPalette={mockResetSelectedPalette}
        resetSelectedProject={mockResetSelectedProject}
        allProjects={mockProjects}
        user={mockUser}
        currentPalette={mockCurrentPalette}
        selectedPaletteInfo={{ name: 'Fake Name', id: 5, project_id: 22 }}
        selectedProjectInfo={mockProjects[2]}
        updateStoredPalette={mockUpdateStoredPalette}
      />)
      wrapper.instance().setState({isUpdateSuccess: true})
    })
    it('should match the alternate snapshot if there is a selectedPalette, and the palette update is a success', () => {
      expect(wrapper).toMatchSnapshot()
    });
  })
})