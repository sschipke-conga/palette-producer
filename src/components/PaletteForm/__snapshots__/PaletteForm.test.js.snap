// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteForm altSnapShot1 should match the alternate snapshot if there is a selectedPalette 1`] = `
<div
  className="palette-form"
>
  <div
    className="update-palette-div"
  >
    <h4>
      <span>
        Palette:
      </span>
       
      Fake Name
    </h4>
    <div
      className="update-palette-buttons"
    >
      <button
        className="update-palette-button"
        onClick={[Function]}
      >
        Update palette
      </button>
      <button
        className="create-palette-button"
        onClick={[Function]}
      >
        Cancel
      </button>
    </div>
  </div>
</div>
`;

exports[`PaletteForm altSnapShot2 should match the alternate snapshot if there is a selectedPalette, and the palette update is a success 1`] = `
<div
  className="palette-form"
>
  <div
    className="update-palette-div"
  >
    <h4>
      <span>
        Palette:
      </span>
       
      Fake Name
    </h4>
    <div
      className="update-palette-buttons"
    >
      <button
        className="update-palette-button"
        onClick={[Function]}
      >
        Update palette
      </button>
      <button
        className="create-palette-button"
        onClick={[Function]}
      >
        Cancel
      </button>
    </div>
    <div
      className="modal-div message"
    >
      <h5>
        Palette successfully updated!
      </h5>
      <button
        onClick={[Function]}
      >
        Return
      </button>
    </div>
  </div>
</div>
`;

exports[`PaletteForm should match the initial snapshot 1`] = `
<div
  className="palette-form"
>
  <form
    className="editor-form"
    onSubmit={[Function]}
  >
    <div
      className="project-editor-div"
    >
      <div
        className="form-label-input-div"
      >
        <label
          htmlFor="password"
        >
          Project
        </label>
        <select
          id="projectId"
          onChange={[Function]}
          required={true}
          value=""
        >
          <option
            defaultValue={true}
            value=""
          >
            Please select a project
          </option>
          <option
            defaultValue={true}
            value="CreateNew"
          >
            Create a new project
          </option>
          <option
            key="Option30"
            value={30}
          >
            My First Project
          </option>
          <option
            key="Option33"
            value={33}
          >
            Second Projects
          </option>
          <option
            key="Option40"
            value={40}
          >
            Third Project
          </option>
        </select>
      </div>
    </div>
    <div
      className="form-label-input-div"
    >
      <label
        htmlFor="paletteName"
      >
        Palette Name
      </label>
      <input
        id="paletteName"
        onChange={[Function]}
        placeholder="Enter a name for your palette"
        required={true}
        type="text"
        value=""
      />
    </div>
    <button
      type="submit"
    >
      Save Palette
    </button>
  </form>
</div>
`;
